app/controllers/api/automate_controller.rb

```diff
@@ -13,22 +13,36 @@ def show
     end
 
     def refresh_from_source_resource(type, id = nil, data = nil)
+      raise BadRequestError, "Must specify an id for refreshing a #{type} resource from source" unless id
+
       api_action(type, id) do |klass|
-        unless GitBasedDomainImportService.available?
-          return action_result(false, 'Please enable the git owner role in order to import git repositories')
-        end
         domain = resource_search(id, type, klass)
-        unless domain.git_repository
-          return action_result(false, "Domain [id=#{id}] did not originate from git repository")
+        api_log_info("Refreshing #{automate_domain_ident(domain)}")
+
+        begin
+          raise "Git owner role is not enabled to import git repositories" unless GitBasedDomainImportService.available?
+          raise "#{automate_domain_ident(domain)} did not originate from git repository" unless domain.git_repository
+          ref = data[:ref] || domain.ref
+          ref_type = data[:ref_type] || domain.ref_type
+
+          description = "Refreshing #{automate_domain_ident(domain)} from git repository"
+          task_id = GitBasedDomainImportService.new.queue_refresh_and_import(domain.git_url,
+                                                                             ref,
+                                                                             ref_type,
+                                                                             current_tenant.id)
+          action_result(true, description, :task_id => task_id)
+        rescue => err
+          action_result(false, err.to_s)
         end
-        ref = data[:ref] || domain.ref
-        GitBasedDomainImportService.new.import(domain.git_repository.id, ref, current_tenant.id)
-        action_result(true, 'Domain refreshed from git repository')
       end
     end
 
     private
 
+    def automate_domain_ident(domain)
+      "Automate Domain id:#{domain.id} fqname:'#{domain.fqname}'"
```

Comment:
@abellotti 
Domains don't have a fqname. Their name itself is the fqname. So domain.name would be the same a domain.fqname